### PR TITLE
Add `snap_getLocale` documentation

### PR DESCRIPTION
**Pull request by ziad-saab**

### CODE REVIEW

**Comment by ziad-saab** [comment]
@SocketSecurity ignore-all

File: snaps/index.mdx [hunk 1]
```diff
@@ -78,6 +78,12 @@ Features include:
       href: "reference/permissions#endowmentnetwork-access",
       title: "Network access",
       description: <>Make API calls using <code>fetch()</code>.</>
+    },
+    {
+      href: "reference/rpc-api#snap_getlocale",
```
Comment (FrederikBolding) [inline snaps/index.mdx:83]: Needs an icon

cc @eriknson

File: package-lock.json
Comment (FrederikBolding) [inline package-lock.json:1]: Is deleting this file intentional?

File: snaps/index.mdx (hunk 1 above)
Comment (Montoya) [reply snaps/index.mdx:83]: He already identified the `home` icon from the DS to be used for the permission, so we just need to find that icon and use it here

Comment (Montoya) [reply snaps/index.mdx:84]: Just pushed an update with the correct icon!

File: src/theme/DocSidebarItem/index.module.css [hunk 2]
```diff
@@ -2,15 +2,9 @@
...
   border-radius: 4px;
   padding: 2px 4px;
   transform: none;
-}
-
-html[data-theme='dark'] .flaskOnly > :global(a.menu__link::after),
-html[data-theme='dark'] .flaskOnly > :global(.menu__list-item-collapsible) > :global(a.menu__link::after) {
-  color: hsl(280, 45%, 85%);
-  background-color: hsl(280, 45%, 25%);
 }
```
Comment (Mrtenz) [inline src/theme/DocSidebarItem/index.module.css:10]:
```suggestion
}

```

**Review by Mrtenz — CHANGES_REQUESTED** [review]

**Review by FrederikBolding — APPROVED** [review]

**Review by FrederikBolding — APPROVED** [review]

File: snaps/reference/rpc-api.md [hunk 3]
```diff
@@ -562,6 +562,40 @@ console.log(entropy);
...
+### snap_getLocale
+
+:::flaskOnly
+:::
+
+Gets the user's locale setting. You can use this method to localize text in your snap.
+
+#### Returns
+
+The user's locale setting.
```
Comment (vandan) [inline snaps/reference/rpc-api.md:574]: Should we provide detail on the specific locale format that's returned here? Is there a reference that can be included?

Comment (alexandratran) [reply snaps/reference/rpc-api.md:574]: @ziad-saab Do we expect the return format to be an ISO two-letter language code? https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes

Comment (Montoya) [reply snaps/reference/rpc-api.md:574]: Hadn't thought about this. What is the format that MetaMask uses?

Comment (Montoya) [reply snaps/reference/rpc-api.md:574]: OK, regardless of format, here's the locales: https://github.com/MetaMask/metamask-extension/blob/develop/app/_locales/index.json

What's the best way you suggest surfacing this information in the doc? Would be happy with a link to this file.

Comment (alexandratran) [inline snaps/reference/rpc-api.md:574]:
```suggestion
The user's locale setting as a [language code](https://github.com/MetaMask/metamask-extension/blob/develop/app/_locales/index.json).
```

Here's a suggestion @Montoya

**Review by alexandratran — APPROVED** [review]
LGTM

**Pull request merged**